adapters/yandex/yandex.go

```diff
@@ -185,21 +203,40 @@ func mapExtToPlacementID(yandexExt openrtb_ext.ExtImpYandex) (*yandexPlacementID
 }
 
 func modifyImp(imp *openrtb2.Imp) error {
+	imp.DisplayManager = bidderName
+	imp.DisplayManagerVer = bidderVersion
+
+	var hasSupportedType bool
+
 	if imp.Banner != nil {
 		banner, err := modifyBanner(*imp.Banner)
-		if banner != nil {
-			imp.Banner = banner
+		if err != nil {
+			return err
 		}
-		return err
+		imp.Banner = banner
+		hasSupportedType = true
+	}
+
+	if imp.Video != nil {
+		video, err := modifyVideo(*imp.Video)
+		if err != nil {
+			return err
+		}
+		imp.Video = video
+		hasSupportedType = true
 	}
 
 	if imp.Native != nil {
-		return nil
+		hasSupportedType = true
 	}
 
-	return &errortypes.BadInput{
-		Message: fmt.Sprintf("Unsupported format. Yandex only supports banner and native types. Ignoring imp id #%s", imp.ID),
+	if !hasSupportedType {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("Unsupported format. Yandex only supports banner, video, and native types. Ignoring imp id #%s", imp.ID),
+		}
 	}
```

Comment:
Can restructure using switch case ? 
```

	switch {
	case imp.Video != nil:
		video, err := modifyVideo(*imp.Video)
		if err != nil {
			return err
		}
		imp.Video = video
		return nil
		
	case imp.Banner != nil:
		banner, err := modifyBanner(*imp.Banner)
		if err != nil {
			return err
		}
		imp.Banner = banner
		return nil
		
	case imp.Native != nil:
		return nil
		
	default:
		return &errortypes.BadInput{
			Message: fmt.Sprintf("Unsupported format. Yandex only supports banner, video, and native types. Ignoring imp id #%s", imp.ID),
		}
	}
```

Reply:
I would not recommend to change `if` cases to `switch` cases. In case of multiformat impression only one type(first case match) will be handled. 
So far it looks like this bidder does not block multiformat imps.